### PR TITLE
[Workers] Allow web workers to be shared

### PR DESCRIPTION
**Pull request by VWoeltjen**
Support an additional flag in the  extension category
such that SharedWorkers may be used. nasa/openmctweb#308.

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Expect to pass code review? Y

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Branch open308search contains an example of this in use to share web worker for generic search (such that domain objects don't get re-indexed if a running instance has already completed this indexing.)

File: docs/src/guide/index.md [hunk 1]
```diff
@@ -1254,6 +1254,21 @@ object, or the current view proxy.
...
+using the `workerService`.
+
+An extension of this category has no implementation. The following properties
+are supported:
+
+* `key`: A symbolic string used to identify this worker.
+* `workerUrl`: The path, relative to this bundle's `src` folder, where
+  this worker's source code resides.
+* `shared`: Optional; a boolean flag which, if true, indicates that this
+  worker should be instantiated as a
```
Comment (larkin) [inline docs/src/guide/index.md]: What's the default value?

File: platform/execution/src/WorkerService.js [hunk 2]
```diff
@@ -66,7 +70,8 @@ define(
          */
```
Comment (akhenry) [inline platform/execution/src/WorkerService.js]: Being picky, but method now returns {Worker | SharedWorker}, so JSDoc probably needs updating. Might be worth mentioning as well when each is returned (ie. that a SharedWorker is specified using the 'shared' flag at declaration time)

**Comment by akhenry** [comment]
Minor jsdoc issue, otherwise looks good!

**Comment by akhenry** [comment]
Assuming minor issue above is addressed -

### Reviewer's Checklist
* Changes appear to address issue? __Y__
* Appropriate unit tests included? __Y__
* Code style and in-line documentation are appropriate? __Y__
* Commit messages meet standards? __Y__

**Pull request merged**